### PR TITLE
fix: unexpected error when handling sign-in callback in React and Vue SDKs

### LINKED ISSUE #508: Bug: Incorrect Return Value from isSignInRedirected in watchEffect
https://github.com/logto-io/js/blob/4f23bf9c2693bd2420c9df22345ef44bd00907a0/packages/vue/src/index.ts#L137-L143

The code provided in the `watchEffect` block has a bug that causes the `isSignInRedirected` function to always return true, regardless of its actual logic. This issue arises because the function `isSignInRedirected` is an async function and is called without awaiting its execution.

### LINKED ISSUE #516: bug: When a page user logs in, an error will occur on the callview page, an infinite loop of printing error messages.
Using logo1.5 and @ logo/vue 2.0.1 versions,
When a user logs in, there is a small probability that an error will occur on the callview page. When checking the control, it is found that there is an infinite loop of printing error messages.
The callback page uses the official recommended writing method

使用logto1.5版本和@logto/vue 2.0.1版本,
用户登录时，小概率会出现在callview页面出错的情况，查看控制发现无限循环打印错误信息。也不给跳出的机会.
callbak page 使用的是官方推荐写法.

这个是引起错误的url
> http://127.0.0.1:5173/callback?error=invalid_target&error_description=resource%20indicator%20must%20be%20an%20absolute%20URI&state=RA8oJLat9V6-TO61udded36nYCOLu3Z0caYFYuok8yAZRKEtw-FL-q5MwvZRCDQxIbxuAm5zUjL_yLCvILeAxg&iss=https%3A%2F%2Faccount.kcpop.plus%2Foidc

image

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Fix potential failure when handling sign-in callback in React and Vue SDKs. Should fix #508 and may also fix #516 

## Testing

UT

## Checklist

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/react/src/hooks/index.ts [hunk 1]
```diff
@@ -82,11 +82,22 @@ const useHandleSignInCallback = (callback?: () => void) => {
...
-      void handleSignInCallback(currentPageUrl);
+    if (!logtoClient) {
+      return;
     }
+
+    (async () => {
+      const currentPageUrl = window.location.href;
+      const isAuthenticated = await logtoClient.isAuthenticated();
+      const isRedirected = await logtoClient.isSignInRedirected(currentPageUrl);
+      console.log('########################## isAuthenticated:', isAuthenticated);
```
Comment (wangsijie) [inline packages/react/src/hooks/index.ts:93]: Forget to remove?

**Review by wangsijie — APPROVED** [review]

File: packages/react/src/hooks/index.ts (hunk 1 above)
Comment (charIeszhao) [reply packages/react/src/hooks/index.ts:93]: Ah, thanks

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.0'
+lockfileVersion: '6.1'
```
Comment (gao-sun) [inline pnpm-lock.yaml:1]: should be 6.0, upgrade your local pnpm

**Review by gao-sun — APPROVED** [review]

File: pnpm-lock.yaml (hunk 2 above)
Comment (charIeszhao) [reply pnpm-lock.yaml:1]: Ah, ok

**Pull request merged**